src/components/layout-components/survey-properties.tsx

```diff
@@ -0,0 +1,145 @@
+import { Box, FormControlLabel, FormGroup, InputAdornment, Switch, TextField, Typography } from "@mui/material"
+import strings from "../../localization/strings"
+import { Survey, SurveyStatus } from "../../generated/client";
+import { ChangeEvent } from "react";
+import WithDebounce from "../generic/with-debounce";
+import TimerOutlinedIcon from "@mui/icons-material/TimerOutlined";
+import { Edit } from "@mui/icons-material";
+import theme from "../../styles/theme";
+
+/**
+ * Component props
+ */
+interface Props {
+  survey: Survey;
+  onSaveSurvey: (event: ChangeEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * Renders Survey Properties component
+ */
+const SurveyProperties = ({ survey, onSaveSurvey }: Props) => {
+
+  const renderWithDebounceTextField = (
+    name: string,
+    onChange: (event: ChangeEvent<HTMLInputElement>) => void,
+    value: string,
+    placeholder: string,
+    endAdornment: boolean
+  ) => (
+    <WithDebounce
+      name={ name }
+      value={ value }
+      onChange={ onChange }
+      placeholder={ placeholder }
+      component={ props =>
+        <TextField
+          { ...props }
+          fullWidth
+          InputProps={{
+            endAdornment: (
+              endAdornment &&
+              <InputAdornment position="end">
+                <Edit fontSize="small" color="primary" />
+              </InputAdornment>
+            )
+          }}
+        />
+      }
+    />
+  );
+
+  const renderWithDebounceNumberField = (
+    name: string,
+    onChange: (event: ChangeEvent<HTMLInputElement>) => void,
+    value: number,
+    placeholder: string
+  ) => (
+    <WithDebounce
+      name={ name }
+      value={ value }
+      onChange={ onChange }
+      placeholder={ placeholder }
+      component={ props =>
+        <TextField
+          { ...props }
+          type="number"
```

Comment:
Are we certain that we want `type="number"`, same result could be applied with `pattern="[0-9]+"`? At least then there wouldn't be those annoying arrow buttons next to the input field 😆

Reply:
I think pattern might not be supported on every browser and those arrows can be removed with some css if needed.